File: CS/WindowsApplication1/Form1.cs
using System;
using System.Data;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            invoicesBindingSource.DataSource = GetInvoicesDataTable();
            colUnitPrice.FieldNameSortGroup = "ExtendedPrice";
            colUnitPrice.SortOrder = DevExpress.Data.ColumnSortOrder.Ascending;
        }

        DataTable GetInvoicesDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Invoices";
            table.Columns.Add(new DataColumn("ProductName", typeof(string)));
            table.Columns.Add(new DataColumn("Discount", typeof(int)));
            table.Columns.Add(new DataColumn("Quantity", typeof(int)));
            table.Columns.Add(new DataColumn("UnitPrice", typeof(double)));
            table.Columns.Add(new DataColumn("ExtendedPrice", typeof(double)));
            Random random = new Random();
            for(int i = 0; i < 20; i++)
            {
                int index = i + 1;
                table.Rows.Add("Product " + index, index * 2, index % 9, (index + random.Next(1, 10)) * 1000d, (index + random.Next(1, 10)) * 1000d + index * 100d);
            }
            return table;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace WindowsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            invoicesBindingSource.DataSource = GetInvoicesDataTable()
            colUnitPrice.FieldNameSortGroup = "ExtendedPrice"
            colUnitPrice.SortOrder = DevExpress.Data.ColumnSortOrder.Ascending
        End Sub

        Private Function GetInvoicesDataTable() As DataTable
            Dim table As DataTable = New DataTable()
            table.TableName = "Invoices"
            table.Columns.Add(New DataColumn("ProductName", GetType(String)))
            table.Columns.Add(New DataColumn("Discount", GetType(Integer)))
            table.Columns.Add(New DataColumn("Quantity", GetType(Integer)))
            table.Columns.Add(New DataColumn("UnitPrice", GetType(Double)))
            table.Columns.Add(New DataColumn("ExtendedPrice", GetType(Double)))
            Dim random As Random = New Random()
            For i As Integer = 0 To 20 - 1
                Dim index As Integer = i + 1
                table.Rows.Add("Product " & index, index * 2, index Mod 9, (index + random.[Next](1, 10)) * 1000.0R, (index + random.[Next](1, 10)) * 1000.0R + index * 100.0R)
            Next
            Return table
        End Function
    End Class
End Namespace
